packages/rich-text/src/plugins/Leaf/index.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { RenderLeafProps } from 'slate-react';
+
+import { Bold } from '../Bold';
+import { Code } from '../Code';
+import { Italic } from '../Italic';
+import { Underline } from '../Underline';
+
+export function Leaf(props: RenderLeafProps) {
+  const { attributes, leaf } = props;
+  let { children } = props;
+
+  if ('bold' in leaf) {
+    children = <Bold {...props}>{children}</Bold>;
+  }
```

Comment:
Good catch Taye. Looking at this again, it seems like it will execute the if-block even if `leaf.bold=false`

I think we should be using what you suggested. Unless I'm missing something too :)